index.js

```diff
@@ -137,11 +143,15 @@ module.exports = fp(function from (fastify, opts, next) {
         this.request.log.warn(err, 'response errored')
         if (!this.sent) {
           if (err.code === 'ERR_HTTP2_STREAM_CANCEL' || err.code === 'ENOTFOUND') {
-            onError(this, { error: new createError.ServiceUnavailable() })
+            onError(this, { error: ServiceUnavailableError() })
           } else if (err instanceof TimeoutError || err.code === 'UND_ERR_HEADERS_TIMEOUT') {
-            onError(this, { error: new createError.GatewayTimeout() })
+            onError(this, { error: new GatewayTimeoutError() })
+          } else if (err.code === 'ECONNRESET') {
+            onError(this, { error: new ConnectionResetError() })
```

Comment:
I created these special cases. Does it make sense?

Reply:
yes, that's ok